src/components/implementation/no_interface/scheddev/sl.c

```diff
@@ -17,21 +17,24 @@ struct sl_global sl_global_data;
  * These functions are removed from the inlined fast-paths of the
  * critical section (cs) code to save on code size/locality
  */
-void
+int
 sl_cs_enter_contention(union sl_cs_intern *csi, union sl_cs_intern *cached, thdcap_t curr, sched_tok_t tok)
 {
 	struct sl_thd *t           = sl_thd_curr();
 	struct sl_global *g        = sl__globals();
+	int ret;
 
 	/* recursive locks are not allowed */
 	assert(csi->s.owner != t->thdcap);
 	if (!csi->s.contention) {
 		csi->s.contention = 1;
-		if (!ps_cas(&g->lock.u.v, cached->v, csi->v)) return;
+		if (!ps_cas(&g->lock.u.v, cached->v, csi->v)) return 1;
 	}
 	/* Switch to the owner of the critical section, with inheritance using our tcap/priority */
-	cos_defswitch(csi->s.owner, t->prio, g->timeout_next, tok);
+	if ((ret = cos_defswitch(csi->s.owner, t->prio, g->timeout_next, tok))) return ret;
```

Comment:
You'll need parenthesis around assignment and conditional, GCC spits out warnings if you don't.
And I don't think this code breaks CSG style.